Controller/Transaction/CommitWebpayM22.php

```diff
@@ -0,0 +1,225 @@
+namespace Transbank\Webpay\Controller\Transaction;
+
+use Transbank\Webpay\Model\TransbankSdkWebpay;
+use Transbank\Webpay\Model\LogHandler;
+
+use \Magento\Sales\Model\Order;
+
+
+/**
+ * Controller for commit transaction Webpay
+ */
+class CommitWebpayM22 extends \Magento\Framework\App\Action\Action {
+
+    private $paymentTypeCodearray = array(
+        "VD" => "Venta Debito",
+        "VN" => "Venta Normal",
+        "VC" => "Venta en cuotas",
+        "SI" => "3 cuotas sin interés",
+        "S2" => "2 cuotas sin interés",
+        "NC" => "N cuotas sin interés",
+    );
+
+    public function __construct(\Magento\Framework\App\Action\Context $context,
+                                \Magento\Checkout\Model\Cart $cart,
+                                \Magento\Checkout\Model\Session $checkoutSession,
+                                \Magento\Framework\Controller\Result\JsonFactory $resultJsonFactory,
+                                \Magento\Framework\Controller\Result\RawFactory $resultRawFactory,
+                                \Transbank\Webpay\Model\Config\ConfigProvider $configProvider) {
+
+        parent::__construct($context);
+
+        $this->cart = $cart;
+        $this->checkoutSession = $checkoutSession;
+        $this->resultJsonFactory = $resultJsonFactory;
+        $this->resultRawFactory = $resultRawFactory;
+        $this->messageManager = $context->getMessageManager();
+        $this->configProvider = $configProvider;
+        $this->log = new LogHandler();
+    }
+
+    /**
+     * @Override
+     */
+    public function execute() {
+
+        $config = $this->configProvider->getPluginConfig();
+        $orderStatusCanceled = $config['error_pay'];
+
+        try {
+
+            $order = $this->getOrder();
+
+            $tokenWs = isset($_POST['token_ws']) ? $_POST['token_ws'] : null;
+
+            if($tokenWs != $this->checkoutSession->getTokenWs()) {
+                throw new \Exception('Token inválido');
+            }
+
+            $paymentOk = $this->checkoutSession->getPaymentOk();
+
+            if ($paymentOk  == 'WAITING') {
+
+                $transbankSdkWebpay = new TransbankSdkWebpay($config);
+                $result = $transbankSdkWebpay->commitTransaction($tokenWs);
+
+                $this->checkoutSession->setResultWebpay($result);
+
+                if (isset($result->buyOrder) && isset($result->detailOutput) && $result->detailOutput->responseCode == 0) {
+
+                    $this->checkoutSession->setPaymentOk('SUCCESS');
+
+                    $authorizationCode = $result->detailOutput->authorizationCode;
+                    $payment = $order->getPayment();
+                    $payment->setLastTransId($authorizationCode);
+                    $payment->setTransactionId($authorizationCode);
+                    $payment->setAdditionalInformation([\Magento\Sales\Model\Order\Payment\Transaction::RAW_DETAILS => (array)$result]);
+
+                    $orderStatus = $config['sucefully_pay'];
```

Comment:
´sucefully_pay´ is this correct?

Reply:
@eugenioLopezRamos yes, this was not changed in this PR